x-pack/legacy/plugins/lens/public/indexpattern_plugin/field_item.tsx

```diff
@@ -41,22 +98,301 @@ export function FieldItem({ field, indexPattern, highlight, exists }: FieldItemP
+          results.topValues &&
+          results.topValues.buckets.length
+        ) {
+          const count = results.sampledValues || results.sampledDocuments || results.totalDocuments;
+          const totalValuesCount =
+            results.topValues &&
+            results.topValues.buckets.reduce((prev, bucket) => bucket.count + prev, 0);
+          const otherCount = count && totalValuesCount ? count - totalValuesCount : 0;
+          // Default to histogram when top values are less than 10% of total
+          setShowingHistogram(otherCount / totalValuesCount > 0.9);
+        } else {
+          setShowingHistogram(!!results.histogram);
+        }
+      })
+      .catch(() => {
+        setState(s => ({ ...s, isLoading: false }));
+      });
+  }
+
+  const specId = getSpecId(
+    i18n.translate('xpack.lens.indexPattern.fieldStatsCountLabel', {
+      defaultMessage: 'Count',
+    })
+  );
+  const expectedColor = getColorForDataType(field.type);
+  const colors: DataSeriesColorsValues = {
+    colorValues: [],
+    specId,
+  };
+  const seriesColors = new Map([[colors, expectedColor]]);
+
+  const bucketedValuesCount =
+    state.topValues && state.topValues.buckets.reduce((prev, bucket) => bucket.count + prev, 0);
+  const otherCount =
+    state.sampledValues && bucketedValuesCount ? state.sampledValues - bucketedValuesCount : 0;
+
+  const euiButtonColor =
+    field.type === 'string' ? 'accent' : field.type === 'number' ? 'secondary' : 'primary';
+  const euiTextColor =
+    field.type === 'string' ? 'accent' : field.type === 'number' ? 'secondary' : 'default';
+
   return (
-    <DragDrop
-      value={{ field, indexPatternId: indexPattern.id } as DraggedField}
-      data-test-subj="lnsFieldListPanelField"
-      draggable
-      className={`lnsFieldListPanel__field lnsFieldListPanel__field-btn-${
-        field.type
-      } lnsFieldListPanel__field--${exists ? 'exists' : 'missing'}`}
+    <EuiPopover
+      id="lnsFieldListPanel__field"
+      display="block"
+      container={document.querySelector<HTMLElement>('.application') || undefined}
+      button={
+        <DragDrop
+          value={{ field, indexPatternId: indexPattern.id } as DraggedField}
+          data-test-subj="lnsFieldListPanelField"
+          draggable
+          className={`lnsFieldListPanel__field lnsFieldListPanel__field-btn-${
+            field.type
+          } lnsFieldListPanel__field--${exists ? 'exists' : 'missing'}`}
+        >
+          <EuiKeyboardAccessible>
+            <div
+              className="lnsFieldListPanel__fieldInfo"
+              data-test-subj={`lnsFieldListPanelField-${field.name}`}
+              onClick={() => {
+                if (exists) {
+                  setOpen(!infoIsOpen);
+                  if (!infoIsOpen) {
+                    fetchData();
+                  }
+                }
+              }}
+              onKeyPress={event => {
+                if (exists && event.key === 'ENTER') {
```

Comment:
I think it would be better to have a helper function that opens the flyout and loads the data, then call that anywhere you're trying to toggle the flyout:

```js
const toggleFlyout = () => {
  if (!exists) {
    return;
  }
  
  setOpen(!infoIsOpen);
  if (!infoIsOpen) {
    fetchData();
   }
};
```

That reduces the likelihood of failing to fetch data when attempting to open the flyout.